app/src/main/java/de/westnordost/streetcomplete/quests/construction/MarkCompletedConstruction.java

```diff
@@ -0,0 +1,80 @@
+package de.westnordost.streetcomplete.quests.construction;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import de.westnordost.osmapi.map.data.Element;
+import de.westnordost.streetcomplete.data.meta.OsmTaggings;
+import de.westnordost.streetcomplete.data.osm.Countries;
+import de.westnordost.streetcomplete.data.osm.OsmElementQuestType;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment;
+import de.westnordost.streetcomplete.quests.DateUtil;
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;
+
+public abstract class MarkCompletedConstruction implements OsmElementQuestType {
+	protected final OverpassMapDataDao overpassServer;
+
+	MarkCompletedConstruction(OverpassMapDataDao overpassServer)
+	{
+		this.overpassServer = overpassServer;
+	}
+
+	@Nullable
+	@Override public Boolean isApplicableTo(Element element)
+	{
+		/* Whether this element applies to this quest cannot be determined by looking at that
+		   element alone (see download()), an Overpass query would need to be made to find this out.
+		   This is too heavy-weight for this method so it always returns false. */
+
+		/* The implications of this are that this quest will never be created directly
+		   as consequence of solving another quest and also after reverting an input,
+		   the quest will not immediately pop up again. Instead, they are downloaded well after an
+		   element became fit for this quest. */
+		return null;
+	}
+
+	public AbstractQuestAnswerFragment createForm()
+	{
+		return new YesNoQuestAnswerFragment();
+	}
+
+	protected String getCurrentDateString(){
+		return DateUtil.getCurrentDateString() + "T00:00:00Z";
+	}
+
+	protected String getOffsetDateString(int offset){
+		return DateUtil.getOffsetDateString(offset) + "T00:00:00Z";
+	}
+
+	String getQueryPart(String key, String nameOfGeneratedGroup, int reviewIntervalInDays){
+		return "[" + key + "=construction]" +
+			"(if:!is_date(t['opening_date']) || date(t['opening_date'])<date('" + getCurrentDateString() + "'))" +
+			" -> .construction_with_unknown_state; " +
+			"way[" + key + "=construction](newer: '" + getOffsetDateString(-reviewIntervalInDays) +"') -> .recently_edited_construction; " +
```

Comment:
There is a bug here.
Here is a hardcoded `way`, even though in `MarkCompletedBuildingConstruction`, you explicitly want to also find relations. Either only do it for ways and leave relations out of it (for simplicity sake) or you have to extend this method further to support this.

Also, if/when an  `older` is introduced, the query could be simplified. Should keep this on a internal todo list or put a TODO in the code here

Reply:
Ouch, good catch.